### PR TITLE
Improve symmetry in client/server address/port text

### PR DESCRIPTION
**Pull request by trask**
Attempt to improve symmetry of
* `(client|server).(address|port)`
* `(client|server).socket.(address|port)`

based on @Oberon00's https://github.com/open-telemetry/semantic-conventions/issues/206#issue-1819655650.

I haven't (intentionally) changed any definitions, so I don't think change log is needed, but not a problem to add.

### CODE REVIEW

**Review by lmolkova — APPROVED** [review]
LGTM!

**Review by AlexanderWert — APPROVED** [review]

File: model/client.yaml
Comment (Oberon00) [inline model/client.yaml:15]: Either we need allow a domain name in the client.address as well, or we forbid it in the server address.

**Review by Oberon00 — CHANGES_REQUESTED** [review]

File: model/client.yaml
Comment (Oberon00) [inline model/client.yaml:20]: (Not related to symmetry, but it seems that the network.transport attribute was moved to trace-specific attributes, and the port is ambiguous if we don't know whether it was UDP or TCP)

File: model/client.yaml
Comment (lmolkova) [reply model/client.yaml:15]: is there a case when we know a client domain name? 
The asymmetry here is caused by technical asymmetry - clients don't have domain names.

At the same time changing this description and allowing domain name (if there is a tech that needs it) would not be a breaking change.

Comment (lmolkova) [reply model/client.yaml:15]: (and if we wanted to set client domain name on a client span, we'd probably prefer  to do it on `host.name` resource attribute rather than `client.address`)

Comment (trask) [reply model/client.yaml:15]: @Oberon00 since this PR isn't changing the status quo, are you ok with merging as-is and we can keep iterating on trying to address all of your concerns expressed in #206?

File: model/client.yaml
Comment (trask) [reply model/client.yaml:20]: can you create an issue for this?

File: model/client.yaml
Comment (Oberon00) [reply model/client.yaml:15]: @lmolkova Host resource attribute is a good point. While a host can have multiple addresses & domain names and there could be a paricular one that is relevant for a particular operation, it definitely reduces the importance. Still, the difference should be called out & justified in the description.

@trask Fair enough

**Comment by trask** [comment]
@open-telemetry/specs-semconv-maintainers can this be merged? I'd like to continue iterating on @Oberon00's concerns from #206

**Review by jsuereth — APPROVED** [review]
Approving assuming @Oberon00's comments are handled in the follow on bug.

File: model/client.yaml
Comment (trask) [reply model/client.yaml:20]: created #250

File: model/client.yaml
Comment (trask) [reply model/client.yaml:15]: created #251

**Pull request merged**
